src/Illuminate/Cache/RedisStore.php

```diff
@@ -444,6 +448,17 @@ protected function pack($value, $connection)
         return $this->serialize($value);
     }
 
+    /**
+     * Determine if the given value should be stored as plain value or be serialized/compressed instead.
+     *
+     * @param  mixed  $value
+     * @return bool
+     */
+    protected function storePlainValue($value): bool
+    {
+        return is_numeric($value) && ! in_array($value, [INF, -INF]) && ! is_nan($value);
```

Comment:
Please don't do this. You will break the cache component when serialization/compression is enabled. I will explain in the issue ticket.

Reply:
Instead, we should ensure that the rate limiter component also is aware of those settings.

Reply:
We can merge this one